### PR TITLE
[Card] Add support for multiple secondary actions in footer and the option to disable them

### PR DESCRIPTION
**Pull request by qq99**
~~Should merge after https://github.com/Shopify/polaris-react/pull/1622 to minimize merge conflicts~~

### WHAT is this pull request doing?

- Adds ability to supply array of `secondaryFooterActions`, and adds deprecation notes to `secondaryFooterAction` (singular)
- Allows `primaryFooterAction` and `secondaryFooterAction` to be disableable

With an array of actions, you can now easily achieve this:
Screen Shot 2019-06-05 at 10 57 17 AM

With an array of just 1 action, it falls back to rendering as if you had supplied a `secondaryFooterAction` (singular):
Screen Shot 2019-06-05 at 10 57 53 AM

Here's a picture of disabling the secondary, and of a custom disclosure text:
Screen Shot 2019-06-05 at 10 58 08 AM

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Card, TextField} from '../src';

interface State {}

/* eslint-disable no-alert */
export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Card
          sectioned
          title="My cool card"
          actions={[{content: 'Header action'}]}
          primaryFooterAction={{content: 'Primary', disabled: true}}
          secondaryFooterAction={{content: 'yo'}}
          secondaryFooterActions={[
            {
              content: 'Secondary 1',
              onAction: () => alert('sec1'),
            },
            {
              content: 'Secondary 2',
              onAction: () => alert('sec2'),
            },
            {
              content: 'Disabled Secondary 3',
              disabled: true,
              onAction: () => alert('sec3'),
            },
          ]}
        >
          <TextField
            label="Textfield"
            readOnly
            disabled
            onChange={() => alert('test')}
          />
        </Card>
        <Card
          title="My cool card"
          actions={[{content: 'Header action'}]}
          primaryFooterAction={{content: 'Primary'}}
          secondaryFooterActions={[
            {
              content: 'Secondary 1',
              onAction: () => alert('sec1'),
            },
          ]}
        >
          <Card.Section>Card body</Card.Section>
          <Card.Section>Card body</Card.Section>
        </Card>
        <Card
          title="Disabled secondary"
          actions={[{content: 'Header action'}]}
          primaryFooterAction={{content: 'Primary'}}
          secondaryFooterActions={[
            {
              content: 'Secondary 1',
              disabled: true,
              onAction: () => alert('sec1'),
            },
          ]}
        >
          <Card.Section>Card body</Card.Section>
          <Card.Section>Card body</Card.Section>
        </Card>
        <Card
          title="Custom disclosure text"
          actions={[{content: 'Header action'}]}
          primaryFooterAction={{content: 'Primary'}}
          secondaryFooterActions={[
            {
              content: 'Secondary 1',
              onAction: () => alert('sec1'),
            },
            {
              content: 'Secondary 2',
              onAction: () => alert('sec2'),
            },
          ]}
          secondaryFooterActionsDisclosureText="..."
        >
          <Card.Section>Card body</Card.Section>
          <Card.Section>Card body</Card.Section>
        </Card>
        {/* Add the code you want to test in here */}
      </Page>
    );
  }
}
/* eslint-enable no-alert */

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: src/components/Card/README.md [hunk 1]
```diff
@@ -266,7 +266,29 @@ Use footer actions for a card’s most important actions, or actions merchants s
 ```jsx
 <Card
   title="Shipment 1234"
-  secondaryFooterAction={{content: 'Cancel shipment'}}
```
Comment (qq99) [inline src/components/Card/README.md]: as the old API is deprecated, I figured I'd update the existing singular example

Not sure why the lines below are showing up as new though

**Comment by alex-page** [comment]
Hey @qq99 I have asked for translations. From a quick glance this looks okay. Have you tophatted this change in mobile, multiple browsers and for accessibility?

**Comment by qq99** [comment]
I could do multiple browsers easily enough

I don't know how to test accessibility though and don't have a good solution to test this on a mobile phone

**Comment by qq99** [comment]
We've been chatting about using `ComplexAction` instead in Slack (cc @chloerice), 
after https://github.com/Shopify/polaris-react/pull/1625 ships I can rebase

File: src/components/Card/Card.tsx [hunk 2]
```diff
@@ -20,16 +27,30 @@ export interface Props {
...
+  /** Secondary actions in the card footer */
+  secondaryFooterActions?: ComplexAction[];
+  /** The text to use for the secondary actions' disclosure button, if a disclosure button is rendered */
+  secondaryFooterActionsDisclosureText?: string;
+}
+
+export type CombinedProps = Props & WithAppProviderProps;
+
+export interface State {
+  secondarFooterActionsPopoverOpen: boolean;
```
[the remark this answers is not in the document]
Comment (qq99) [reply src/components/Card/Card.tsx]: :o nice catch

File: src/components/Card/tests/Card.test.tsx [hunk 3]
```diff
@@ -92,6 +92,110 @@ describe('<Card />', () => {
...
+          <p>Some card content.</p>
+        </Card>,
+      );
+
+      const button = card.find(Button).first();
+      expect(button).toHaveLength(1);
+      expect(button.text()).toBe('a');
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (qq99) [reply src/components/Card/tests/Card.test.tsx]: There is in a way, in `popover works when disclosure button is clicked`:

```tsx
      const disclosureButton = card.find(Button).first();
      expect(disclosureButton).toHaveLength(1);
      expect(disclosureButton.text()).toBe('More');
```

**Comment by qq99** [comment]
I think I got all your changes @chloerice 🤞

**Pull request merged**